src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -214,8 +208,31 @@ public override bool SendChunked
         /// <returns>Returns the value of the header with the given name.</returns>
         public override string GetHeader(string headerName)
         {
-            Util.CheckArgumentNullAndEmpty(headerName, "headerName");
-            return HttpWebRequestMessage.GetHeaderValue(this.httpRequest, headerName);
+            if (_contentHeaderNames.Contains(headerName))

```

Comment:
Try and write unit tests for this method to verify that header can be fetched from `_requestMessage.Content.Headers`, `_contentHeaderValueCache` and `_requestMessage.Headers` as the case may be. Otherwise its hard to confirm that it behaves as expected

Reply:
there are tests showing this.